packages/flutter/lib/src/widgets/form.dart

```diff
@@ -607,19 +607,19 @@ class FormFieldState<T> extends State<FormField<T>> with RestorationMixin {
 
   @override
   Widget build(BuildContext context) {
+    final FormState? formState = Form.maybeOf(context);
     if (widget.enabled) {
-      switch (widget.autovalidateMode) {
-        case AutovalidateMode.always:
+      final AutovalidateMode? formMode = formState?.widget.autovalidateMode;
+      final AutovalidateMode fieldMode = widget.autovalidateMode;
+      if(formMode == AutovalidateMode.always || fieldMode == AutovalidateMode.always) {
+        _validate();
+      } else if(formMode == AutovalidateMode.onUserInteraction || fieldMode == AutovalidateMode.onUserInteraction) {
```

Comment:
Same question about precedence here, too.

Reply:
About those missing spaces, should these be notified by default in the IDE (VS Code in my case) or is there a command to run this locally before committing? I didn't find anything specifically about it in the docs but maybe it's an issue with my setup 🤔

Reply:
My editor doesn't catch this kind of thing either, and this repo doesn't use the Dart autoformatter so that's not an option. I've noticed a few things like this that are only caught by the analyzer after pushing the PR and not by the lint rules. Maybe there's a way to check everything in the analyzer in the IDE, and if so I wish I knew it!